features/src/autogluon/features/generators/datetime.py

```diff
@@ -67,10 +79,19 @@ def _generate_features_datetime(self, X: DataFrame) -> DataFrame:
             # from other features.
             for feature in self.features:
                 X_datetime[datetime_feature + '.' + feature] = getattr(X_datetime[datetime_feature].dt, feature).astype(int)
-
-            X_datetime[datetime_feature] = pd.to_numeric(X_datetime[datetime_feature])
-
+            
+            if holiday_encoding == True:
+                for i in range(len(X_datetime)):
+                    X_datetime[datetime_feature + "_holiday"].iloc[i] = int(date(X_datetime[datetime_feature + ".year"].iloc[i], X_datetime[datetime_feature + ".month"].iloc[i], X_datetime[datetime_feature + ".day"].iloc[i]) in holidays_list)
+                X_datetime[datetime_feature] = pd.to_numeric(X_datetime[datetime_feature])
+            
+            if sincos_encoding == True: # add Encoding cyclical continuous features
+    # https://ianlondon.github.io/blog/encoding-cyclical-features-24hour-time/
+                X_datetime[datetime_feature + '.dayofweek'] = np.sin(2 * np.pi * X_datetime[datetime_feature + '.dayofweek'] / 7.0) 
+                X_datetime[datetime_feature + '.dayofweek'] = np.cos(2 * np.pi * X_datetime[datetime_feature + '.dayofweek'] / 7.0)
```

Comment:
We are just overwriting the value here? Isn't the idea to have 2 features computed from this? They should have different names. Perhaps:

```
X_datetime[datetime_feature + '.dayofweek_sin'] = np.sin(2 * np.pi * X_datetime[datetime_feature + '.dayofweek'] / 7.0) 
X_datetime[datetime_feature + '.dayofweek_cos'] = np.cos(2 * np.pi * X_datetime[datetime_feature + '.dayofweek'] / 7.0)
X_datetime = X_datetime.drop(columns=[datetime_feature + '.dayofweek'])
```